core/rawdb/freezer.go

```diff
@@ -342,8 +342,18 @@ func (f *Freezer) TruncateHead(items uint64) (uint64, error) {
 	if oitems <= items {
 		return oitems, nil
 	}
-	for _, table := range f.tables {
-		if err := table.truncateHead(items - f.offset); err != nil {
+	for kind, table := range f.tables {
+		err := table.truncateHead(items - f.offset)
+		// if truncate below tail, try reset to it
+		if err == errTruncationBelowTail {
```

Comment:
can we do resetItems for sidecars
other table keep as before？
It's a more safe way